google/cloud/spanner/integration_tests/data_types_integration_test.cc

```diff
@@ -63,35 +65,43 @@ StatusOr<T> WriteReadData(Client& client, T const& data,
   return actual;
 }
 
-class DataTypeIntegrationTest
-    : public spanner_testing::DatabaseIntegrationTest {
- public:
+template <typename T>
```

Comment:
Did you consider composition instead of inheritance?  I am guessing you did. Can you add a comment explaining why it does not work?

Reply:
GoogleTest fixtures are all based on inheriting from `testing::Test`.  It looks to me like it would be more difficult (or at least less familiar) to buck that trend.

Reply:
I was thinking along these lines:

```cc
template <typename T>
class DataTypeIntegrationTest : public ::testing::Test {
 private:
   T base_impl_;
};
```

Reply:
Yeah, maybe, I guess.  If `T` isn't a `testing::Test` it would mean agreeing on some new interface.  But if it is, I'm not sure that it is buying you much.

Reply:
Up to you, I don't need to look at this code that often.